### ISSUE TITLE
Empty --to-prefix is either ignored or treated as "true"

### ISSUE THREAD

**Issue by PierreBeucher** [generalui/s3p#73]
When running `sync` (probably with `cp` and such as well), specifying empty `--to-prefix` to remove a source `--prefix` is either ignored or treated as `'true'` depending how parameter is passed.

Consider a bucket with content:

```
s3://source/some/prefix/file.txt
```

I want to copy into target such as:

```
s3://target/file.txt
```

---
I tried with:

```
s3p sync --bucket s3://source --prefix "some/prefix" --to-bucket s3://target --to-prefix ""
```

But `--to-prefix ""` is ignored like it hasn't been passed. Result is:

```
s3://target/some/prefix/file.txt
```

---

Setting with `=` somehow convert prefix as `'true'` string (probably because [empty string is truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) in JS)

```
s3p sync --bucket s3://source --prefix "some/prefix" --to-bucket s3://target --to-prefix=""
```

Result in:

```
s3://target/true/file.txt
```

---

I worked around using a custom JS function such as:

```
s3p sync --bucket s3://source --prefix "some/prefix" --to-bucket s3://target \
  --to-key "js:(key) => key.replace('some/prefix/', '')" 
```

But it would be nice to have a working `--to-prefix` :)

**Comment by shanebdavis** [comment]
Nice find. I agree this is a bug. Until I have a fix, something to try in the mean time: I'll bet using `--to-prefix="js:''"` will work.

Details: An empty `--to-prefix` is automatically set to `true` by the CLI pre-processor, so that tracks. Clearly the empty string is triggering the classic JavaScript bug of empty-string == false.

**Comment by shanebdavis** [comment]
Fixed in v3.4.2
